### PR TITLE
Issue/5949 th warnings

### LINKED ISSUE #5949: [Tests] - console.warn empty table headers
With one of the updates in the test libraries, a console warn crept its way unseen. It is an accessibility warning, nothing breaking atm. 

image

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

Remove the errors about the accesibility of empty Table Headers - Service Inventory has still some warning, but as redesign will remove most of the tables from that view

closes #5949 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Comment by LukasStordeur** [comment]
Considering you are applying the changes to ISO-7, please also resolve the warnings in the Service Inventory. It shouldn't be much more work I believe.

File: src/UI/words.tsx [hunk 1]
```diff
@@ -812,6 +812,7 @@ const dict = {
   "common.compileWidget.compiling": "Compiling",
   "common.compileWidget.compilationDisabled.hint":
     "The server_compile setting is disabled. You can enable it on the settings page under the configuration tab.",
+  "common.emptyColumnHeader": "Empty Column Header",
```
Comment (LukasStordeur) [inline src/UI/words.tsx:815]: Considering this is for accessiblity reasons, it should state what the collumn is about. 
```suggestion
  "common.emptyColumnHeader": "Column header for collapsible-expandable action",
```

File: src/Slices/CompileDetails/UI/CompileStageReportTable.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const CompileStageReportTable: React.FC<Props> = ({
     <Table {...props} variant={TableVariant.compact}>
       <Thead>
         <Tr>
-          <Th aria-hidden key="toggle" />
+          <Th aria-hidden key="toggle" aria-label="tableHeader-toggle" />
```
Comment (LukasStordeur) [inline src/Slices/CompileDetails/UI/CompileStageReportTable.tsx:40]: Why is this one without screenReaderText?

**Review by LukasStordeur — CHANGES_REQUESTED** [review]

**Comment by matborowczyk** [comment]
Screenshot 2024-11-05 at 15 11 09

File: changelogs/unreleased/5949-th-warnings.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+description: Add screen reader text for empty columns headers to improve accessibility
+issue-nr: 5942
```
Comment (LukasStordeur) [inline changelogs/unreleased/5949-th-warnings.yml:2]: I think you made a typo in the ticket-number

File: src/Slices/ServiceInventory/UI/InventoryTable.tsx [hunk 4]
```diff
@@ -70,6 +71,11 @@ export const InventoryTable: React.FC<Props> = ({
         key={column.displayName}
         {...sortParams}
         {...(column.apiName === "actions" && { "aria-hidden": true })}
+        screenReaderText={
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/InventoryTable.tsx:74]: You can simply add the attribute in the line above like so:

```TS
{...(column.apiName === "actions" && { 
  "aria-hidden": true, 
  "screenReaderText": words("common.emptyColumnHeader") 
})}
```

File: src/Slices/CompileDetails/UI/CompileStageReportTable.tsx [hunk 5]
```diff
@@ -37,7 +38,11 @@ export const CompileStageReportTable: React.FC<Props> = ({
     <Table {...props} variant={TableVariant.compact}>
       <Thead>
         <Tr>
-          <Th aria-hidden key="toggle" />
+          <Th
+            aria-hidden
```
Comment (LukasStordeur) [inline src/Slices/CompileDetails/UI/CompileStageReportTable.tsx:42]: just found out that the aria-hidden isn't needed when adding the screenReaderText, no need to make the changes, just an FIY

**Review by LukasStordeur — APPROVED** [review]

**Pull request closed without merging**